### PR TITLE
python3Packages.ehtim: init at 1.2.4

### PR DESCRIPTION
**Pull request by phiadaarr**
Add the python package ehtim to nixpkgs. It is widely used in the Event Horizon Telescope community.

### CODE REVIEW

File: pkgs/development/python-modules/ehtim/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    rev = "v${version}";
+    sha256 = "ZeacOUve1QHX4Ei61GHEMtbqyfOV7Cj9q7+sH/lQR+w=";
+  };
+  patches = [ ./silence.patch ];
+
+  propagatedBuildInputs = [
+    numpy scipy astropy matplotlib ephem h5py pandas requests future
+  ];
+
+  doCheck = false;
```
Comment (azahi) [inline pkgs/development/python-modules/ehtim/default.nix:22]: Why is the check phase disabled?

File: pkgs/development/python-modules/ehtim/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  stdenv, lib, buildPythonPackage, fetchFromGitHub,
+  numpy, scipy, astropy, matplotlib, ephem, h5py, pandas, requests, future
```
Comment (azahi) [inline pkgs/development/python-modules/ehtim/default.nix:3]: Please format these like so:
```
{ lib
, stdenv
, foo
, bar
, ...
}:
```

File: pkgs/development/python-modules/ehtim/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    numpy scipy astropy matplotlib ephem h5py pandas requests future
+  ];
+
+  doCheck = false;
+  pythonImportsCheck = [ "ehtim" ];
+
+  meta = with lib; {
+    homepage = "https://achael.github.io/eht-imaging";
+    description = "Python modules for simulating and manipulating VLBI data and producing images with regularized maximum likelihood methods.";
+    license = licenses.gpl3;
```
Comment (azahi) [inline pkgs/development/python-modules/ehtim/default.nix:28]: https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

File: pkgs/development/python-modules/ehtim/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  pname = "ehtim";
+  version = "1.2.4";
+
+  src = fetchFromGitHub {
+    owner = "achael";
+    repo = "eht-imaging";
+    rev = "v${version}";
+    sha256 = "ZeacOUve1QHX4Ei61GHEMtbqyfOV7Cj9q7+sH/lQR+w=";
+  };
+  patches = [ ./silence.patch ];
```
Comment (azahi) [inline pkgs/development/python-modules/ehtim/default.nix:16]: Are there any other reasons to patch this out apart from making it less noisy? If not, we should leave it as-is, patching should be used primarily for fixing upstream bugs or making sure stuff works on a target platform.

File: pkgs/development/python-modules/ehtim/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  propagatedBuildInputs = [
+    numpy scipy astropy matplotlib ephem h5py pandas requests future
+  ];
+
+  doCheck = false;
+  pythonImportsCheck = [ "ehtim" ];
+
+  meta = with lib; {
+    homepage = "https://achael.github.io/eht-imaging";
+    description = "Python modules for simulating and manipulating VLBI data and producing images with regularized maximum likelihood methods.";
```
Comment (azahi) [inline pkgs/development/python-modules/ehtim/default.nix:27]:
```suggestion
    description = ''
      Python modules for simulating and manipulating VLBI data and
      producing images with regularized maximum likelihood methods
    '';
```

**Review by azahi — CHANGES_REQUESTED** [review]
Please consult [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) and use the PR [template](https://github.com/NixOS/nixpkgs/blob/master/.github/PULL_REQUEST_TEMPLATE.md).

File: pkgs/development/python-modules/ehtim/default.nix (hunk 2 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ehtim/default.nix]: Done.

File: pkgs/development/python-modules/ehtim/default.nix (hunk 4 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ehtim/default.nix]: The reason was that it is not common that python packages print a greeting when being imported. But it is probably something that I should rather discuss with upstream.

File: pkgs/development/python-modules/ehtim/default.nix (hunk 5 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ehtim/default.nix]: Done.

File: pkgs/development/python-modules/ehtim/default.nix (hunk 1 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ehtim/default.nix:31]: Upstream does not have many tests and virtually all of them require additional data. I cannot even run the tests myself. Also something that needs to be discussed with upstream.

**Comment by phiadaarr** [comment]
###### Description of changes

Add the python package ehtim to nixpkgs. It is widely used in the Event Horizon Telescope community.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

File: pkgs/development/python-modules/ehtim/default.nix (hunk 1 above)
Comment (azahi) [reply pkgs/development/python-modules/ehtim/default.nix:31]: Please add a comment saying something like `# Tests require additional data.` if that's the case.

File: pkgs/development/python-modules/ehtim/default.nix (hunk 4 above)
Comment (azahi) [reply pkgs/development/python-modules/ehtim/default.nix]: I don't think that this is a compelling reason to patch this behavior out by default, I believe we should keep upstream's intentions in this regard. A user can always patch that locally if needed.

File: pkgs/development/python-modules/ehtim/default.nix (hunk 1 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ehtim/default.nix]: Done.

**Comment by phiadaarr** [comment]
Is there any more I can do here?

**Review by azahi — APPROVED** [review]

**Pull request closed without merging**